.github/workflows/go.yml

```diff
@@ -1,123 +0,0 @@
-# Copyright 2021 Google LLC
-#
-# Licensed under the Apache License, Version 2.0 (the "License");
-# you may not use this file except in compliance with the License.
-# You may obtain a copy of the License at
-#
-#      http://www.apache.org/licenses/LICENSE-2.0
-#
-# Unless required by applicable law or agreed to in writing, software
-# distributed under the License is distributed on an "AS IS" BASIS,
-# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-# See the License for the specific language governing permissions and
-# limitations under the License.
-
-name: Go
-
-on:
-  push:
-    branches:
-    - main
-  pull_request:
-    branches:
-    - main
-jobs:
-  build:
-    name: Build
-    runs-on: ubuntu-22.04
-    services:
-      postgres:
-        image: postgres:15
-        env:
-          # Define a super user. Password for super user is required.
-          POSTGRES_USER: postgres
-          POSTGRES_PASSWORD: postgres
-          # Allow other connections without a password.
-          POSTGRES_HOST_AUTH_METHOD: trust
-        # Set health checks to wait until postgres has started.
-        options: >-
-          --health-cmd pg_isready
-          --health-interval 10s
-          --health-timeout 5s
-          --health-retries 5
-        ports:
-          # Map tcp service container port 5432 to the host.
-          - 5432:5432
-    steps:
-    - name: Set up Go 1.x
-      uses: actions/setup-go@v2
-      with:
-        go-version: ^1.18
-
-    - name: Check out code into the Go module directory
-      uses: actions/checkout@v2
-
-    - name: golangci-lint
-      uses: golangci/golangci-lint-action@v2
-      with:
-        version: v1.45
-        args: --timeout 5m0s
-
-    - name: Install
-      run: go install ./...
-
-    - name: Install protoc (needed for registry upload tests)
-      uses: arduino/setup-protoc@v1
-      with:
-        repo-token: ${{ secrets.GITHUB_TOKEN }}
-
-    - name: Create default configuration that uses a local server
-      run: registry config configurations create local --registry.address='127.0.0.1:8080' --registry.insecure
-
-    - name: Test everything with SQLite
-      run: go test ./...
-
-    - name: Configure PostgreSQL
-      env:
-        # Connect to the locally mapped port.
-        PGHOST: localhost
-        PGPORT: 5432
-        # Use the super user for configuration.
-        PGUSER: postgres
-        PGPASSWORD: postgres
-      # Create the database and user required by the PostgreSQL tests.
-      run: psql -c "CREATE USER registry_tester" -c "CREATE DATABASE registry_test WITH OWNER registry_tester" 
-
-    - name: Test registry server with PostgreSQL
```

Comment:
I think we discussed dropping this to nightly, but if we are parallelizing tests, how much harm is there in doing this on pushes? (I do think we need to be careful about resource usage during testing and not just runtime, but this PR is making structural changes and test content changes, which makes it harder to follow)

Reply:
Right. That's why I was asking if you wanted to leave this. I can definitely move it back.